pandas/tests/tslibs/test_parsing.py

```diff
@@ -154,27 +154,25 @@ def test_parsers_month_freq(date_str, expected):
         ("2011-12-30T00:00:00", "%Y-%m-%dT%H:%M:%S"),
         ("2011-12-30T00:00:00UTC", "%Y-%m-%dT%H:%M:%S%Z"),
         ("2011-12-30T00:00:00Z", "%Y-%m-%dT%H:%M:%S%z"),
-        # The +9 format for offsets is supported by dateutil,
-        # but don't round-trip, see https://github.com/pandas-dev/pandas/issues/48921
-        ("2011-12-30T00:00:00+9", None),
-        ("2011-12-30T00:00:00+09", None),
+        ("2011-12-30T00:00:00+9", "%Y-%m-%dT%H:%M:%S%z"),
+        ("2011-12-30T00:00:00+09", "%Y-%m-%dT%H:%M:%S%z"),
```

Comment:
This is nice! In

https://github.com/pandas-dev/pandas/blob/113bdb37694e2849a15c390fddd8f28986bf1863/pandas/_libs/tslibs/parsing.pyx#L1003-L1007

we check that `array_strptime` can parse the first non-null element with the guessed format. Now that `array_strptime` can parse both ISO and non-ISO formats, we're expanding on the list of formats which can be guessed!